config/secure-headers.php

```diff
@@ -3,319 +3,515 @@
+
+	/**
+	 * Cross-Origin-Resource-Policy.
+	 *
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Resource-Policy
 	 *
-	 * Reference: https://w3c.github.io/webappsec-clear-site-data/
+	 * Available Value: 'same-site', 'same-origin', 'cross-origin'
 	 */
+	'cross-origin-resource-policy' => 'cross-origin',
 
+	/**
+	 * Clear-Site-Data.
+	 *
+	 * @see https://w3c.github.io/webappsec-clear-site-data/
+	 */
 	'clear-site-data' => [
 		'enable' => false,
 		'all' => false,
 		'cache' => true,
+		'clientHints' => true,
 		'cookies' => true,
 		'storage' => true,
 		'executionContexts' => true,
 	],
 
-	/*
-	 * HTTP Strict Transport Security
+	/**
+	 * HTTP Strict Transport Security.
 	 *
-	 * Reference: https://developer.mozilla.org/en-US/docs/Web/Security/HTTP_strict_transport_security
+	 * @see https://developer.mozilla.org/en-US/docs/Web/Security/HTTP_strict_transport_security
 	 *
 	 * Please ensure your website had set up ssl/tls before enable hsts.
 	 */
-
 	'hsts' => [
 		'enable' => env('SECURITY_HEADER_HSTS_ENABLE', false),
 		'max-age' => 15552000,
 		'include-sub-domains' => false,
+		'preload' => false,
+	],
+
+	/**
+	 * Reporting Endpoints.
+	 *
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Reporting-Endpoints
+	 *
+	 * Note: The array key is the endpoint name, and the value is the URL.
+	 */
+	'reporting' => [
+		// 'csp' => 'https://example.com/csp-reports',
+		// 'nel' => 'https://example.com/nel-reports',
 	],
 
-	/*
-	 * Expect-CT
+	/**
+	 * Network Error Logging.
 	 *
-	 * Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect-CT
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Network_Error_Logging
+	 * @see https://developer.mozilla.org/en-US/docs/Web/API/Reporting_API
 	 */
+	'nel' => [
+		'enable' => false,
+		// The name of reporting API, not the endpoint URL.
+		'report-to' => '',
+		'max-age' => 86400,
+		'include-subdomains' => false,
+		'success-fraction' => 0.0,
+		'failure-fraction' => 1.0,
+	],
 
+	/**
+	 * Expect-CT.
+	 *
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect-CT
+	 */
 	'expect-ct' => [
 		'enable' => false,
 		'max-age' => 2147483648,
 		'enforce' => false,
 		'report-uri' => null,
 	],
 
-	/*
-	 * Public Key Pinning
+	/**
+	 * Feature Policy.
 	 *
-	 * Reference: https://developer.mozilla.org/en-US/docs/Web/Security/Public_Key_Pinning
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy
 	 *
-	 * hpkp will be ignored if hashes is empty.
+	 * 'none', '*' and 'self allow' are mutually exclusive,
+	 * the priority is 'none' > '*' > 'self allow'.
 	 */
+	'permissions-policy' => [
+		'enable' => true,
 
-	'hpkp' => [
-		'hashes' => [
-			// 'sha256-hash-value',
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy/accelerometer
+		'accelerometer' => [
+			'none' => false,
+			'*' => false,
+			'self' => true,
```

Comment:
I won't add for all of these, but I don't think we use many of these, so should probably change to `none`.

Reply:
the one i am wondering about is the auth key one.